src/improvements/tasks/MultisigTask.sol

```diff
@@ -68,105 +58,108 @@ abstract contract MultisigTask is Test, Script, StateOverrideManager, TaskManage
-    function approveFromChildMultisig(string memory taskConfigFilePath, address _childMultisig, bytes memory signatures)
-        public
-    {
+    /// @notice Approves the the root safe transaction from a nested safe.
+    function approve(string memory taskConfigFilePath, address _childMultisig, bytes memory signatures) public {
         // TODO: Remove this when the interface tasks an array of safes.
         address[] memory childSafes = Solarray.addresses(_childMultisig);
-        _taskSetup(taskConfigFilePath, childSafes);
+        (TaskPayload memory payload,) = _taskSetup(taskConfigFilePath, childSafes);
 
-        Action[] memory actions = build();
-        approve(_childMultisig, signatures, actions);
+        // TODO: in the future this index may be different. Any safe that is not the root safe will be eligible to approve.
+        uint256 childSafeIndex = 0;
+        execute(signatures, payload, childSafeIndex);
         console.log(
             "--------- Successfully %s Child Multisig %s Approval ---------",
-            isBroadcastContext() ? "Broadcasted" : "Simulated",
+            _isBroadcastContext() ? "Broadcasted" : "Simulated",
             _childMultisig
         );
     }
 
-    /// @notice Simulates a nested multisig task with the given configuration file path for a
-    /// given child multisig. Prints the 'data to sign' which is used to sign with the eip712sign binary.
-    function signFromChildMultisig(string memory taskConfigFilePath, address _childMultisig)
-        public
-        returns (VmSafe.AccountAccess[] memory, Action[] memory, bytes32, bytes memory)
-    {
-        return simulateRun(taskConfigFilePath, "", _childMultisig);
-    }
-
-    /// @notice This function performs the same functionality as signFromChildMultisig but
-    /// to keep the terminal output clean, we don't want to return the account accesses and actions.
-    function simulateAsSigner(string memory taskConfigFilePath, address _childMultisig) public {
-        simulateRun(taskConfigFilePath, "", _childMultisig);
-    }
-
-    /// @notice Child multisig approves the task to be executed.
-    function approve(address _childMultisig, bytes memory signatures, Action[] memory actions) public {
-        bytes memory approveCalldata = generateApproveMulticallData(actions);
-        bytes32 hash = keccak256(getEncodedTransactionData(_childMultisig, approveCalldata));
-        signatures = Signatures.prepareSignatures(_childMultisig, hash, signatures);
-
-        execTransaction(_childMultisig, MULTICALL3_ADDRESS, 0, approveCalldata, Enum.Operation.DelegateCall, signatures);
-    }
-
-    /// @notice Executes the task with the given signatures.
-    function execute(bytes memory signatures, Action[] memory actions)
+    /// @notice Executes a task for a given safe. The 'safeIndex' is the index of the safe in the payload.safes array.
+    function execute(bytes memory signatures, TaskPayload memory payload, uint256 safeIndex)
```

Comment:
It is quite confusing that the execution path for `execute` in nested.just is:
```
execute(taskConfigFilePath, signatures) public
    _runTask(...) internal
        execute(signatures, payload, safeIndex) public
```

Maybe they should be renamed as:
```
executeFromPath(taskConfigFilePath, signatures) public
executePayload(payload, signatures, safeIndex) public
```
Note I moved arguments around, so that they are first what ot execute, then the signatures, then modifier arguments.